lib/sidekiq/redeploy/loader.rb

```diff
@@ -62,12 +63,12 @@ def process_loop
             reload_app { deployer.deploy(source: deployer.archive_file) }
           elsif reload_sidekiq
             reload_app
-          elsif process_died?(@sidekiq_pid)
-            fork_sidekiq
           end
+          num_dead = process_died?(@sidekiq_pids)
```

Comment:
See my other comment about renaming `process_died?` to `running_pids`

Then we can do something like this:
```
 @sidekiq_pids = running_pids(@sidekiq_pids)
 fork_sidekiq(@num_processes - @sidekiq_pids.length) if @sidekiq_pids.length < @num_processes
```